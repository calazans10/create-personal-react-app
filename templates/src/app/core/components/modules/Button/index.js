import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './index.scss';

const Button = ({ type, children, onClick }) => {
  const btnClasses = classNames({
    button: true,
    [`button--${type}`]: true,
  });

  return (
    <button type="button" className={btnClasses} onClick={onClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(['default', 'primary', 'success', 'danger', 'warning']),
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  type: 'default',
};

export default Button;
