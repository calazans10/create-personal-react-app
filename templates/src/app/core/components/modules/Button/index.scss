@import 'styles/utilities';

.button {
  display: block;
  width: 100%;
  margin-bottom: 0;
  padding: rem(8) rem(4);
  border: none;
  border-radius: rem(5);
  border-color: transparent;
  background-image: none;
  color: #fff;
  font-size: em(14);
  font-weight: normal;
  line-height: 1.3333333;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  touch-action: manipulation;
  user-select: none;

  &--default {
    background-color: #bdc3c7;
  }

  &--primary {
    background-color: #00aeef;
  }

  &--success {
    background-color: #21d376;
  }

  &--danger {
    background-color: #e33b15;
  }

  &--warning {
    background-color: #f39c12;
  }
}
