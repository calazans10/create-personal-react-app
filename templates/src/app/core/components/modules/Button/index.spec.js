import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './index';

describe('Button', () => {
  it('renders without crashing', () => {
    const props = {
      type: 'default',
      children: 'Detalhes',
      onClick: jest.fn(),
    };
    const wrapper = shallow(<Button {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
