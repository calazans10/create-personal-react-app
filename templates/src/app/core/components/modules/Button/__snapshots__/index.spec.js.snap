// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<button
  className="button button--default"
  onClick={[MockFunction]}
  type="button"
>
  Detalhes
</button>
`;
